### DISCUSSION TITLE
Generating all types as public

### DISCUSSION THREAD

**Discussion by vron** [sbrunner/jsonschema-gentypes#640] [Q&A]
Hi,

Thanks for a nice library. I am struggling to find a way to make it generate all the types from a schema as public types (i.e. without a leading underscore) - is this not a supported use-case?

I have a JSON schema with 100s of types (generated from rust code) and want to "export" all those types to python - unfortunately only my top-level type in the schema is generated without a leading _.

Is there any support for making jsonschema-gentypes generate all definitions as public types (without having to list them all explicitly in the name-mapping field - i.e. I want all types (or a regexp in the name mapping) so as to not have to keep a long list of types up to date manually all the time.

/V

**Comment by sbrunner** [comment]
I don't remember why I do that, for me, it can be done without any option :-)

**Reply by vron** [reply]
Hi - I'm not sure i understand what you are saying - how are you able to do that without any option?

E.g. if I run jsonschema-gentypes on:


`{
  "$schema": "http://json-schema.org/draft-07/schema#",
  "title": "Both",
  "type": "object",
  "required": [
    "input",
    "output"
  ],
  "properties": {
    "input": {
      "$ref": "#/definitions/Deal"
    },
    "output": {
      "$ref": "#/definitions/Deal"
    }
  },
  "definitions": {
    "Deal": {
      "type": "object",
      "oneOf": [
        {
          "type": "object",
          "required": [
            "maturity"
          ],
          "properties": {
            "maturity": {
              "type": "string",
              "format": "date"
            }
          }
        }
      ],
      "required": [
        "agio"
      ],
      "properties": {
        "agio": {
          "type": "number",
          "format": "double"
        }
      }
    }
  }
}`

I get:


`
from typing import TypedDict, Union
from typing_extensions import Required


class Both(TypedDict, total=False):
    """ Both. """

    input: Required["_Deal"]
    """
    Aggregation type: oneOf

    Required property
    """

    output: Required["_Deal"]
    """
    Aggregation type: oneOf

    Required property
    """



_Deal = Union["_DealOneof0"]
""" Aggregation type: oneOf """



class _DealOneof0(TypedDict, total=False):
    maturity: Required[str]
    """
    format: date

    Required property
    """

`

I.e the types is named _Deal and not Deal ?

**Reply by sbrunner** [reply]
Yes, It's what I understand, then a pull request that removes this leading underscore will be accepted :-)
